code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -171,7 +175,27 @@
 		force = 30
 	..()
 
+/obj/item/tool/shovel/etool/attackby(obj/item/I, mob/user, params)
+	if(!I.sharp)
+		return ..()
+	if(sharp)
+		to_chat(user, "<span class='notice'>The entrenching tool is already sharpened.</span>")
+		return
+	if(folded)
+		to_chat(user, "<span class='notice'>You cannot sharpen the entrenching tool while it is folded.</span>")
+		return
+	user.visible_message("<span class='notice'>[user] begins to sharpen the [src] with the [I].</span>",
+	"<span class='notice'>You begin to sharpen the [src] with the [I].</span>")
+	if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY))
+		return
+	sharp = TRUE
```

Comment:
not a boolean.  Here are the different values it can take
![image](https://user-images.githubusercontent.com/25491240/114016651-5b8fb780-986b-11eb-9904-bb95ba94529b.png)

Reply:
sharp = IS_SHAR_ITEM_SIMPLE for example. The rest of the code doesn't have to change